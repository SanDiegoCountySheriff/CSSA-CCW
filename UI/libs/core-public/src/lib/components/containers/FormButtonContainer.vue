<template>
  <div class="form-btn-container">
    <v-btn
      color="success mr-2"
      @click="handleSubmit"
      :disabled="!valid"
    >
      {{ $t('Continue') }}
    </v-btn>
    <!-- TODO: this needs to save to save the current state to local storage or call the api to save in
      the db
       -->

    <v-btn color="secondary mr-2">
      {{ $t('Save and Exit') }}
    </v-btn>

    <!-- TODO: Make this return to the home page with out saving the form at all -->
    <v-btn color="error mr-2">
      {{ $t('Cancel') }}
    </v-btn>
  </div>
</template>

<script lang="ts">
import { defineComponent } from "vue";
import { boolean } from "@storybook/addon-knobs";

export default defineComponent({
  name: 'FormButtonContainer',
  props: {
    handleValid: {
      type: boolean,
      default: false,
    },
  },
  methods: {
    handleSubmit() {
      this.$emit('submit');
    },
  },
});
</script>

<style lang="scss" scoped>
.form-btn-container {
  width: 90%;
  margin-top: 0.5rem;
  display: flex;
  justify-content: flex-end;
}
</style>
