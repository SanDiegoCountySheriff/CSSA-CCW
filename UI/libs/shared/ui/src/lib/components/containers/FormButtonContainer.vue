<template>
  <v-card-actions>
    <v-btn
      v-if="
        !props.isFinalStep ||
        (!applicationStore.completeApplication.application
          .isUpdatingApplication &&
          !props.loading)
      "
      :disabled="!props.valid || props.loading || !props.allStepsComplete"
      :loading="props.loading"
      @click="handleContinue"
      color="primary"
    >
      {{ $t('Continue') }}
    </v-btn>
    <v-btn
      color="primary"
      @click="handleSave"
      :disabled="props.loading"
      :loading="props.loading"
    >
      {{ $t('Save and Exit') }}
    </v-btn>
  </v-card-actions>
</template>

<script setup lang="ts">
import { useCompleteApplicationStore } from '@shared-ui/stores/completeApplication'

const applicationStore = useCompleteApplicationStore()

interface FormButtonContainerProps {
  valid?: boolean
  loading?: boolean
  allStepsComplete?: boolean
  isFinalStep?: boolean
}

const props = withDefaults(defineProps<FormButtonContainerProps>(), {
  valid: false,
  loading: false,
  allStepsComplete: true,
  isFinalStep: false,
})

const emit = defineEmits(['continue', 'save'])

function handleContinue() {
  emit('continue')
}

function handleSave() {
  emit('save')

  if (applicationStore.completeApplication.application.isUpdatingApplication) {
    applicationStore.completeApplication.application.isUpdatingApplication =
      false
  }
}
</script>
