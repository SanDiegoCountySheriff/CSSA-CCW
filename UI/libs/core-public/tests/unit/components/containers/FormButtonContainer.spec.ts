/* eslint-disable @typescript-eslint/ban-ts-comment */
import FormButtonContainer from '@shared-ui/components/containers/FormButtonContainer.vue'
import Vuetify from 'vuetify'
import { createLocalVue, mount } from '@vue/test-utils'

const localVue = createLocalVue()
const tMock = {
  $t: text => text,
}

describe('FormButtonContainer', () => {
  let vuetify
  let wrapper

  beforeEach(() => {
    vuetify = new Vuetify()
    //@ts-ignore
    wrapper = mount(FormButtonContainer, {
      localVue,
      vuetify,
      mocks: tMock,
      propsData: {
        valid: true,
      },
    })
  })
  afterEach(() => {
    wrapper.destroy()
  })

  it('should match the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render all three buttons', () => {
    expect(wrapper.findAll('button')).toHaveLength(4)
  })

  it('Should emit on button press', () => {
    const button = wrapper.findAll('button').at(0)

    button.trigger('click')
    expect(wrapper.emitted().submit).toHaveLength(1)
  })
})
