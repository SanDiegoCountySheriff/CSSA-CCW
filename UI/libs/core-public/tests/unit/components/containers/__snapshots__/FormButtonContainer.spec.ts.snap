// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormButtonContainer should match the snapshot 1`] = `
<div class="form-btn-container"><button type="button" class="v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--default success "><span class="v-btn__content">
    Continue
  </span></button> <button type="button" class="v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--default secondary mx-2"><span class="v-btn__content">
    Save and Exit
  </span></button> <button type="button" class="v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--default error mr-2"><span class="v-btn__content">
    Cancel
  </span></button></div>
`;
